### PR TITLE
DAOS-12435 control: Remove dmg storage query target-health cmd

### PR DESCRIPTION
**Pull request by tanabarr**
For MD-on-SSD target information and data can be spread across
multiple SSDs on multiple tiers, as such the output of dmg storage
query target-health command may be less useful because it will include
multiple SSDs including those that either contain meta data or data
for the given target. Since this command was mainly added for
development purposes and seems not to be widely used, remove the
command to reduce unnecessary code and command surface.

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by kjacque — DISMISSED** [review]

**Review by knard-intel — DISMISSED** [review]
LGTM

**Review by NiuYawei — COMMENTED** [review]
I suppose this query request will be sent to engine when engine is running? So should we remove the 'target' parameter from drpc and drpc handler as well?

**Comment by tanabarr** [comment]
> I suppose this query request will be sent to engine when engine is running? So should we remove the 'target' parameter from drpc and drpc handler as well?

The protobuf field is removed here (and the shims are rebuilt to remove references in the auto compiled C and Go files) https://github.com/daos-stack/daos/pull/12031/files#diff-0d2c0a8bd54b4226333cddc7d4563ba0863478dcc8faed08916afdfaa5889e46L129 . This field is unused in any of the dRPC callers or handlers because the implementation (not written by me) calls smd_list_devs without any filter and filters response in the control-plane server https://github.com/daos-stack/daos/pull/12031/files#diff-a533d0aed30e0e5bb5124bf429a67b182cb6709e92845ac3395638b089fde393L97 . So in summary all the relevant code has been removed. Sound okay?

**Comment by NiuYawei** [comment]
> > I suppose this query request will be sent to engine when engine is running? So should we remove the 'target' parameter from drpc and drpc handler as well?
> 
> The protobuf field is removed here (and the shims are rebuilt to remove references in the auto compiled C and Go files) https://github.com/daos-stack/daos/pull/12031/files#diff-0d2c0a8bd54b4226333cddc7d4563ba0863478dcc8faed08916afdfaa5889e46L129 . This field is unused in any of the dRPC callers or handlers because the implementation (not written by me) calls smd_list_devs without any filter and filters response in the control-plane server https://github.com/daos-stack/daos/pull/12031/files#diff-a533d0aed30e0e5bb5124bf429a67b182cb6709e92845ac3395638b089fde393L97 . So in summary all the relevant code has been removed. Sound okay?

I see. I was looking into the ds_mgmt_bio_health_query() which still has a "tgt" argument, looks it's not related to this change. @wangshilong could take a look to see if C code related to this function needs be adjusted?

**Comment by wangshilong** [comment]
> > > I suppose this query request will be sent to engine when engine is running? So should we remove the 'target' parameter from drpc and drpc handler as well?
> > 
> > 
> > The protobuf field is removed here (and the shims are rebuilt to remove references in the auto compiled C and Go files) https://github.com/daos-stack/daos/pull/12031/files#diff-0d2c0a8bd54b4226333cddc7d4563ba0863478dcc8faed08916afdfaa5889e46L129 . This field is unused in any of the dRPC callers or handlers because the implementation (not written by me) calls smd_list_devs without any filter and filters response in the control-plane server https://github.com/daos-stack/daos/pull/12031/files#diff-a533d0aed30e0e5bb5124bf429a67b182cb6709e92845ac3395638b089fde393L97 . So in summary all the relevant code has been removed. Sound okay?
> 
> I see. I was looking into the ds_mgmt_bio_health_query() which still has a "tgt" argument, looks it's not related to this change. @wangshilong could take a look to see if C code related to this function needs be adjusted?

I think Niu means TgtId in BioHealthReq will be unused now.
type BioHealthReq struct {
        state         protoimpl.MessageState
        sizeCache     protoimpl.SizeCache
        unknownFields protoimpl.UnknownFields

        DevUuid string `protobuf:"bytes,1,opt,name=dev_uuid,json=devUuid,proto3" json:"dev_uuid,omitempty"`
        TgtId   string `protobuf:"bytes,2,opt,name=tgt_id,json=tgtId,proto3" json:"tgt_id,omitempty"`
}

tgt id arg for ds_mgmt_bio_health_query shall be removed too, but yup, this PR is OK, i could remove it in a following pr though.

**Comment by tanabarr** [comment]
> > > > I suppose this query request will be sent to engine when engine is running? So should we remove the 'target' parameter from drpc and drpc handler as well?
> > > 
> > > 
> > > The protobuf field is removed here (and the shims are rebuilt to remove references in the auto compiled C and Go files) https://github.com/daos-stack/daos/pull/12031/files#diff-0d2c0a8bd54b4226333cddc7d4563ba0863478dcc8faed08916afdfaa5889e46L129 . This field is unused in any of the dRPC callers or handlers because the implementation (not written by me) calls smd_list_devs without any filter and filters response in the control-plane server https://github.com/daos-stack/daos/pull/12031/files#diff-a533d0aed30e0e5bb5124bf429a67b182cb6709e92845ac3395638b089fde393L97 . So in summary all the relevant code has been removed. Sound okay?
> > 
> > 
> > I see. I was looking into the ds_mgmt_bio_health_query() which still has a "tgt" argument, looks it's not related to this change. @wangshilong could take a look to see if C code related to this function needs be adjusted?
> 
> I think Niu means TgtId in BioHealthReq will be unused now. tgt id arg for ds_mgmt_bio_health_query shall be removed too, but yup, this PR is OK, i could remove it in a following pr though.

yes those are unrelated but I will remove the relevant code in this PR as I can see what you mean now

**Comment by tanabarr** [comment]
@NiuYawei @wangshilong tgt_id removed from bio_health_query in Commit https://github.com/daos-stack/daos/pull/12031/commits/bd745abdeaf972856e1d97d6a39673eeaa56a9b2

**Review by knard-intel — COMMENTED** [review]
It seems that the latest modifications of `src/proto/ctl/smd.proto` file have not been pushed: remove of the `BioHealthReq.TgtId` field.

**Comment by tanabarr** [comment]
> It seems that the latest modifications of `src/proto/ctl/smd.proto` file have not been pushed: remove of the `BioHealthReq.TgtId` field.

thanks, good catch

**Review by knard-intel — APPROVED** [review]
LGTM

**Review by kjacque — APPROVED** [review]
LGTM

**Review by NiuYawei — APPROVED** [review]

**Review by wangshilong — APPROVED** [review]
Thanks for cleanup this.

**Comment by tanabarr** [comment]
CI failure, known issue due to DAOS-12725, @daos-stack/daos-gatekeeper can we please force land this PR

File: src/bio/README.md [hunk 1]
```diff
@@ -179,12 +179,10 @@ Pools
 ```
 
 <a id="81"></a>
-- Query Device Health Data: **$dmg storage query (device-health | target-health)**
+- Query Device Health Data: **$dmg storage query (device-health**
```
Comment (mjmac) [inline src/bio/README.md:182]: nit: Unbalanced parens

**Review by mjmac — APPROVED** [review]

File: src/bio/README.md (hunk 1 above)
Comment (tanabarr) [reply src/bio/README.md:182]: Thanks Mike, I will correct in the next PR targeted at master.

Comment (tanabarr) [reply src/bio/README.md:182]: corrected in https://github.com/daos-stack/daos/pull/12079

**Pull request merged**